nemo/core/classes/mixins/adapter_mixins.py

```diff
@@ -159,6 +182,14 @@ def add_adapter(self, name: str, cfg: DictConfig):
             cfg: A DictConfig or Dataclass that contains at the bare minimum `__target__` to instantiate a
                 new Adapter module.
         """
+        if hasattr(self, '_accepted_adapter_types'):
```

Comment:
Instead of directly accessing the variables l, use `get_accepted_adapter_types(self,)` to get the list and then check length.